contracts/OwnedSet.sol

```diff
@@ -103,10 +98,10 @@ contract OwnedSet is Owned, ValidatorSet {
 	function addValidator(address _validator)
 		external
 		onlyOwner
-		isNotPending(_validator)
+		isNotValidator(_validator)
 	{
-		pendingStatus[_validator].isIn = true;
-		pendingStatus[_validator].index = pending.length;
+		status[_validator].isIn = true;
```

Comment:
As discussed, additional check can be made off-chain, since the contract only emits events anyway.

Maybe let's just add a big fat comment to `report*` functions and explicitly state that in case someone takes that contract and attempts to modify it.